Fix for sdk25

rjaros87 opened the pull request.

rjaros87 commented:
@brantone , @gildegoma please review. Fix for https://github.com/gildegoma/chef-android-sdk/issues/34

rjaros87 committed:
Add support for SDK 25

rjaros87 committed:
comment out 'wip suite'

Brantone commented:
Seems like lots of other peripheral changes (ex: remove tailor from Rakefile), and bunch of white space changes, which make it somewhat cumbersome to isolate actual changes to SDK25.
Possible to split out the actual changes for  gildegoma#34 SDK25 and extras? Or even maybe few extra comments for other un-referenced chanages?

Brantone commented on Rakefile:
```diff
@@ -2,40 +2,13 @@
+task default: [:rubocop, :foodcritic, :knife, :serverspec]
 
 desc 'Lint Ruby code'
-task :tailor do
```
Why removal?

rjaros87 replied:
Rubocop checks the the same and even more.

rjaros87 replied:
and also we execute foodcritic ;)

rjaros87 commented on .kitchen.yml:
```diff
@@ -8,88 +8,97 @@ provisioner:
-      jdk_version: 7
-
-- name: default
+- name: legacy
```
Suite to check the backward compatibility

rjaros87 commented on .kitchen.yml:
```diff
@@ -8,88 +8,97 @@ provisioner:
         - android-22
         - sys-img-armeabi-v7a-android-22
+
+- name: default
```
Suite to check the _latest_ SDK and default recipe without modification attributes `android-sdk`

rjaros87 commented on attributes/default.rb:
```diff
@@ -1,18 +1,27 @@
+default['android-sdk']['checksum']['mac_os_x']        = '593544d4ca7ab162705d0032fb0c0c88e75bd0f42412d09a1e8daa3394681dc6'
+default['android-sdk']['checksum']['windows']         = '23d5686ffe489e5a1af95253b153ce9d6f933e5dbabe14c494631234697a0e08'
+
+if ::Gem::Version.new(node['android-sdk']['version']) < ::Gem::Version.new('25')
```
We check the SDK version and set `node['android-sdk']['legacy_sdk']` attribute which will be used in our recipies. Also there are different urls.

rjaros87 commented on recipes/unix.rb:
```diff
@@ -71,6 +73,7 @@
   owner node['android-sdk']['owner']
   group node['android-sdk']['group']
   backup node['android-sdk']['backup_archive']
+  strip_components node['android-sdk']['legacy_sdk'] ? 1 : 0
```
Major fix for unix system is just simply extract archive to `android-sdk` directory without strip directory (by default ark enter to the first directory and then extract the content (default value: `1`))

rjaros87 commented on recipes/windows.rb:
```diff
@@ -43,9 +43,15 @@
-    ::File.rename "#{setup_root}/android-sdk-windows", "#{setup_root}/#{node['android-sdk']['name']}"
+    if node['android-sdk']['legacy_sdk']
+      ::File.rename "#{setup_root}/android-sdk-windows", "#{setup_root}/#{node['android-sdk']['name']}"
+    else
```
Major fix for windows system

rjaros87 commented:
Extras are only `Rakefile`, `Gemfile` and `.rubocop`

`.kitchen.yml` Had a bad indentation and strangely combined attributes. Of course, I had to add tests for SDK 25 and SDK 24 

`.travis.yml` I had problem with Ubuntu machine, because they don't see package urls `404` - `apt-get update` solves the problem and by the way I raised versions of ruby to be the same as is in the ChefDK

`Berksfile` (test-kitchen) the same problems as was on Travis

`default.rb` and recipies - see my comments above

`test/integration/*` - because now we have two suites (default and legacy) I used shared context to not duplicate scenario :)

`dna.json` - just update sdk version

Hope my comments will help.

gildegoma commented:
Thank you *a lot* and *again* @rjaros87 and @Brantone to collaborate on this issue.
I'd like to precise that gildegoma#34 is a bug fix issue for the `0.2.x` version of the cookbook, and its fix must absolutely be *focused* and be based on cb67ebd25d656f47fb581ed9a81c9d11c50abb86 (`0.2.0` without any additional changes). The backwards compatible approach of auto-detected `node['android-sdk']['legacy_sdk']` looks great at first glance. I'd really love to receive a PR that only address the issue for the Linux platform (instead of being based on https://github.com/gildegoma/chef-android-sdk/pull/33 stuff). The work for version `0.3.0` is a quite larger topic, where more refactorings are fully allowed.

❤️ ❤️ ❤️

rjaros87 committed:
[GH-34] Fix broken installation of SDK Tools 25+

gildegoma committed:
Bump and reorganize FoodCritic

In order to prepare a more significant switch to ChefDK, the FoodCritic
exceptions are now defined in .foodcritic, and no longer in the
Rakefile.

Enable back FC041 (download are done via the ark cookbook only)

Tolerate FC053 as the 'recommends' keyword is still worth to be present,
even though Chef tools are not using this meta data.

gildegoma committed:
Update CHANGELOG and README according to v0.2.1

gildegoma committed:
[GH-34] Always download .zip packages with new URL

After four random checks, it seems that the new HTTPS-based
repository provides all the Linux package history. The archives are in
.zip format, and don't require to strip an `android-sdk-linux` base
directory.

Since the new .zip archives only contain the 'tools' directory, it is
no longer necessary to fix the permissions and ownership for
the 'add-ons' or 'platforms' that were present in the .tgz file.

gildegoma committed:
v0.2.1 QA: Update Vagrantfile and .kitchen.yml

rjaros87 committed:
Merge branch 'master' of github.com:rjaros87/chef-android-sdk into windows_fix_sdk25

rjaros87 committed:
Update test and refactor Unix recipe

rjaros87 commented:
Merged with upstream master :) and refactor unix recipe

rjaros87 commented on recipes/unix.rb:
```diff
@@ -71,17 +73,10 @@
-#
-# Fix non-friendly 0750 permissions in order to make android-sdk available to all system users
-#
-%w(add-ons platforms tools).each do |subfolder|
```
we don't need this anymore :) by default in archive `tools` has `755`
